src/Identity/Extensions.Core/src/IdentityBuilder.cs

```diff
@@ -32,7 +38,14 @@ public IdentityBuilder(Type user, IServiceCollection services)
     /// <param name="role">The <see cref="Type"/> to use for the roles.</param>
     /// <param name="services">The <see cref="IServiceCollection"/> to attach to.</param>
     public IdentityBuilder(Type user, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] Type role, IServiceCollection services) : this(user, services)
-        => RoleType = role;
+    {
+        if (role.IsValueType)
```

Comment:
Is this a breaking change?

Reply:
`RoleType` and `UserType` are later used to generate types, all of which have a `where T : class` constraint.

Throwing here means the error is different and it is thrown in a different place. But a value type here never worked.